src/main/java/gift/model/Option.java

```diff
@@ -53,4 +54,12 @@ public Long getQuantity() {
     public Product getProduct() {
         return product;
     }
+
+    public void subtract(Long quantity) {
+        Long remainingQuantity = this.quantity - quantity;
+        if (remainingQuantity < 0) {
+            throw new OutOfStockException("해당 옵션의 수량이 부족합니다.");
+        }
+        this.quantity -= quantity;
```

Comment:
```suggestion
        this.quantity = remainingQuantity;
```

굳이 두번 계산할 필요가 없을거 같아서 변수 추출을 이야기 드린거였습니다.